Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2338,11 +2337,39 @@ static NSInteger _colCompare(id block1, id block2, void *)
 
 #endif
 
+
+static String preferredFilenameForElement(const HTMLImageElement& element)
```

Comment:
What's the output for an `<img>` that is not backed by an `<attachment>`, and has a blob URL?

Reply:
I'm not sure; the non-attachment logic is loosely based on how `attributedString` decides on a file name. How do I make a test page that has an image with a blob url?

Reply:
Easiest way is probably to just copy an image from Notes (or another system application) into a `contenteditable` in Safari / Minibrowser.

Otherwise, create a Blob using JS, `URL.createObjectURL`, and assign to an `<img>`'s src.